spark-datasource-core/build.gradle.kts

```diff
@@ -0,0 +1,49 @@
+plugins {
+    id("scala")
+    id("scala-conventions")
+    id("publishing-conventions")
+}
+
+description = "Salesforce Data Cloud Spark DataSource core implementation"
+val mavenName: String by extra("Salesforce Data Cloud Spark DataSource Core")
+val mavenDescription: String by extra("${project.description}")
+
+dependencies {
+    // ALIGNED WITH JDBC-CORE PATTERN: Use compileOnly for gRPC
+    compileOnly(project(":jdbc-grpc"))
+    compileOnly(libs.grpc.stub)
+    compileOnly(libs.grpc.protobuf)
+    
+    // Core implementation dependencies (like jdbc-core)
+    implementation(project(":jdbc-util"))
+    implementation(libs.slf4j.api)
+    
+    // Spark dependencies
+    implementation("org.apache.spark:spark-sql_2.13:3.5.5")
+    implementation("org.apache.spark:spark-core_2.13:3.5.5")
```

Comment:
let's move these dependencies into `gradle/libs.versions.toml` so that the versions are always in sync - we shouldn't have any direct dependencies in `build.gradle.kts` unless we have a good reason